catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -193,13 +203,27 @@ public Metacard getMetacard(String id)
         }
 
         Filter queryFilter = filterBuilder.anyOf(filters);
-        QueryResponse response = catalogFramework.query(new QueryRequestImpl(new QueryImpl(
-                queryFilter,
-                1,
-                -1,
-                SortBy.NATURAL_ORDER,
-                false,
-                TimeUnit.SECONDS.toMillis(10)), false));
+        QueryRequest queryRequest;
+
+        if (CollectionUtils.isEmpty(writableSources)) {
+            queryRequest = new QueryRequestImpl(new QueryImpl(
+                    queryFilter,
+                    1,
+                    querySize,
```

Comment:
@lessarderic If this PR hadn't been open for 2 months, I would say yes. It will likely be more effective to create a ticket and address it in a smaller, separate PR.